lib/engine/game/g_steam_over_holland/game.rb

```diff
@@ -356,25 +363,40 @@ def non_floated_corporations
         end
 
         def check_distance(route, visits, train = nil)
-          super
-
           raise GameError, 'Route cannot begin/end in a town' if visits.first.town? || visits.last.town?
+
+          super
         end
 
         def revenue_for(route, stops)
-          super
+          revenue = super
+
+          abilities = abilities(route.corporation, :hex_bonus)
+          return revenue if abilities.empty?
 
-          raise GameError, 'Route visits same hex twice' if route.hexes.size != route.hexes.uniq.size
+          bonus_hexes = abilities.flat_map(&:hexes)
+          bonus = 20 * stops.map(&:hex).map(&:coordinates).intersect(bonus_hexes).size
+          revenue + bonus
         end
 
         def sell_shares_and_change_price(bundle, allow_president_change: true, swap: nil, movement: nil)
           price_drop = bundle.num_shares
           corporation = bundle.corporation
           old_price = corporation.share_price
+          old_owner = bundle.owner
+          was_president = corporation.president?(bundle.owner)
```

Comment:
You've just saved this value in a variable, so you might as well not look it up again.
```suggestion
          was_president = corporation.president?(old_owner)
```

Reply:
very good point! done